py/gpu_specter/extract/cpu.py

```diff
@@ -498,13 +506,10 @@ def ex2d_patch(noisyimg, imgweights, A4, decorrelate='signal'):
     #- Resolution matrix (B&S eq 12)
     R = np.outer(1.0/norm_vector, np.ones(norm_vector.size)) * Q
 
-    #- Decorrelated covariance matrix (B&S eq 13-15)
-    Cx = R.dot(Cov.dot(R.T))
-    
     #- Decorrelated flux (B&S eq 16)
     fx = R.dot(f.ravel()).reshape(f.shape)
     
-    #- Variance on f (B&S eq 13)
-    varfx = np.diagonal(Cx).reshape(fx.shape)
+    #- Inverse variance on f (B&S eq 13)
+    ivarfx = (norm_vector**2).reshape(fx.shape)
```

Comment:
Good optimization, not bothering to do `R.dot(Cov.dot(R.T))` just to create a diagonal matrix to extract its diagonal.

Reply:
Good catch, I forgot that I snuck that in here. I was wondering if there was a reason for returning var instead of ivar since it's immediately inverted anyway. The other motivation I had for this was to align with return signature of specter's ex2d_patch.